Source/Scene/Cesium3DTileset.js

```diff
@@ -1286,309 +1470,276 @@ define([
-                        // are ready. This allows the visible children to stay loaded while other children stream in,
-                        // otherwise if only the parent is selected the subtree may be unloaded from the cache.
-                        selectTile(tileset, t, fullyVisible, frameState);
-
-                        for (k = 0; k < childrenLength; ++k) {
-                            child = children[k];
-                            if (isVisible(child.visibilityPlaneMask)) {
-                                if (child.contentUnloaded) {
-                                    requestContent(tileset, child, outOfCore);
-                                } else {
-                                    stack.push(child);
-                                }
-                            }
-                        }
-                    } else {
-                        // Tile does not meet SSE and its visible children are loaded. Refine to them in front-to-back order.
-                        for (k = 0; k < childrenLength; ++k) {
-                            child = children[k];
-                            if (isVisible(child.visibilityPlaneMask)) {
-                                stack.push(child);
-                            }
-                        }
+        queueTile(tileset, start, start, stack, selectionState, frameState, outOfCore);
 
-                        t.replaced = true;
-                        if (defined(t.descendantsWithContent)) {
-                            scratchRefiningTiles.push(t);
-                        }
+        while (stack.length > 0) {
+            var tile = stack.pop();
+            visitTile(tileset, tile, frameState, outOfCore);
+            queueTile(tileset, start, tile, stack, selectionState, frameState, outOfCore);
+        }
+    }
+
+    /**
+     * Load and add a tile to the proper queue and load and push children to the processing stack
+     */
+    function queueTile(tileset, start, tile, stack, selectionState, frameState, outOfCore) {
+        var nextQueue = selectionState.nextQueue;
+        var finalQueue = selectionState.finalQueue;
+        var maximumScreenSpaceError = tileset._maximumScreenSpaceError;
+        var loadSiblings = tileset.loadSiblings;
+
+        if (tile.hasTilesetContent) {
+            if (tile.contentUnloaded) {
+                tile._loadHeap.insert(tile);
+            }
+            if (!tile.contentReady) {
+                finalQueue.push(tile);
+            } else {
+                var child = tile.children[0];
+                child.visibilityPlaneMask = tile.visibilityPlaneMask;
+                child.updateTransform(tile.computedTransform);
+                child.distanceToCamera = tile.distanceToCamera;
+                stack.push(child);
+            }
+            return;
+        }
+
+        if (tile.refine === Cesium3DTileRefine.ADD) {
+            loadAndAddToQueue(tileset, tile, finalQueue);
```

Comment:
It checks `tile.visibilityPlaneMask` which is this tile's visibility. This has been computed because `queueTile` is only called on tiles that have been added to the queue. Either `computeChildrenVisibility` or `updateAndPushChildren` is called on a tile before adding a tile to the nextQueue.